core/templates/pages/splash-page/splash-page.component.ts

```diff
@@ -29,47 +30,74 @@ const splashConstants = require('constants.ts');
 angular.module('oppia').component('splashPage', {
   template: require('./splash-page.component.html'),
   controller: [
-    '$rootScope', '$timeout', 'LoaderService', 'SiteAnalyticsService',
-    'UrlInterpolationService', 'UserService', 'WindowRef',
+    '$rootScope', 'LoaderService', 'SiteAnalyticsService',
+    'UrlInterpolationService', 'UserService', 'WindowDimensionsService',
     function(
-        $rootScope, $timeout, LoaderService, SiteAnalyticsService,
-        UrlInterpolationService, UserService, WindowRef) {
+        $rootScope, LoaderService, SiteAnalyticsService,
+        UrlInterpolationService, UserService, WindowDimensionsService) {
       var ctrl = this;
       ctrl.getStaticImageUrl = function(imagePath) {
         return UrlInterpolationService.getStaticImageUrl(imagePath);
       };
-      ctrl.getStaticSubjectImageUrl = function(subjectName) {
-        return UrlInterpolationService.getStaticImageUrl(
-          '/subjects/' + subjectName + '.svg');
-      };
-
-      ctrl.onRedirectToLogin = function(destinationUrl) {
-        SiteAnalyticsService.registerStartLoginEvent(
-          'splashPageCreateExplorationButton');
-        $timeout(function() {
-          WindowRef.nativeWindow.location = destinationUrl;
-        }, 150);
-        return false;
-      };
 
       ctrl.onClickBrowseLessonsButton = function() {
         SiteAnalyticsService.registerClickBrowseLessonsButtonEvent();
-        $timeout(function() {
-          WindowRef.nativeWindow.location = (
-            `/learn/${splashConstants.DEFAULT_CLASSROOM_URL_FRAGMENT}`);
-        }, 150);
         return false;
       };
 
-      ctrl.onClickCreateExplorationButton = function() {
-        SiteAnalyticsService.registerClickCreateExplorationButtonEvent();
-        $timeout(function() {
-          WindowRef.nativeWindow.location = '/creator-dashboard?mode=create';
-        }, 150);
-        return false;
+      ctrl.getMainBackgroundUrl = function() {
+        if (ctrl.isWindowNarrow()) {
+          return ctrl.getStaticImageUrl('/splash/splashMainMobile.png');
+        } else {
+          return ctrl.getStaticImageUrl('/splash/splashMainDesktop.png');
+        }
+      };
+
+      ctrl.isWindowNarrow = function() {
+        return WindowDimensionsService.isWindowNarrow();
+      };
+
+      // The 2 functions below are to cycle between values: 0, 1, 2 for
```

Comment:
Better to generalize this by having a constant for the total number of testimonials, since we may add more in the future and you don't want to have to change that number everywhere.

Reply:
Done